### ISSUE TITLE
Ore map not seed deterministic between restarts

### ISSUE THREAD

**Issue by rbneiman** [anegostudios/VintageStory-Issues#3557]
### Game Version

v1.19.3

### Platform

Windows

### Modded

Vanilla

### SP/MP

None

### Description

I don't know if this is intentional or not, but the generated ore maps are not the same between worlds with the same seed, but only if you restart the game between generating worlds.

Just because I was curious, I looked at the [vssurvivalmod](https://github.com/anegostudios/vssurvivalmod) repo and I noticed that the ``getOrCreateMapLayer()`` function in ``GenDeposits.cs`` uses the ``String.GetHashCode()`` function on the ore names which is not deterministic on strings between restarts.

### How to reproduce

1. Generate a world, use a prop. pick on a block.
2. Generate the world again using the same seed then use a prop. pick on the same block as before. The ore readings are the same.
3. Restart the game
4. Generate the world again using the same seed then use a prop. pick on the same block as before. The ore readings are now completely different.

### Screenshots

Initial world:
![2024-02-04_18-13-36](https://github.com/anegostudios/VintageStory-Issues/assets/24992900/6fc697f0-061d-4483-8a76-a714b2718a76)
Another world with same seed, no restart (readings are the same):
![2024-02-04_18-15-20](https://github.com/anegostudios/VintageStory-Issues/assets/24992900/a808f723-fc06-41ed-900d-a8bb9cabbccf)
Another world with same seed, after restart (readings are completely different):
![2024-02-04_18-43-11](https://github.com/anegostudios/VintageStory-Issues/assets/24992900/6cce6ab7-8a3e-47b6-8e50-cf9fd698c465)

### Logs

I inserted a print statement in the ``getOrCreateMapLayer()`` function to verify that the hashcodes were actually changing on restart.

Run 1:
```Log
4.2.2024 17:43:31 [Event] Starting world generators...
4.2.2024 17:43:31 [Warning] world seed: 891551837 mapcode: diamond hashcode: 806444314
4.2.2024 17:43:31 [Warning] world seed: 891551837 mapcode: emerald hashcode: -1174349482
4.2.2024 17:43:31 [Warning] world seed: 891551837 mapcode: peridot hashcode: -1310198435
```

Run2 (after restart):
```Log
4.2.2024 17:44:46 [Event] Starting world generators...
4.2.2024 17:44:46 [Warning] world seed: 891551837 mapcode: diamond hashcode: -650860141
4.2.2024 17:44:46 [Warning] world seed: 891551837 mapcode: emerald hashcode: -2036280494
4.2.2024 17:44:46 [Warning] world seed: 891551837 mapcode: peridot hashcode: -1094736333
...
```

**Comment by radfast** [comment]
@rbneiman this is an awesome issue report, thank you for the detective work on the string hashcodes

**Comment by radfast** [comment]
Fixed in upcoming 1.19.5, thank you again for the most helpful issue report.

**Comment by rbneiman** [comment]
No problem!
